modules/signals/src/with-hooks.ts

```diff
@@ -16,13 +16,23 @@ type HooksFactory<Input extends SignalStoreFeatureResult> = (
   >
 ) => void;
 
-export function withHooks<Input extends SignalStoreFeatureResult>(hooks: {
+type HooksSupplier<Input extends SignalStoreFeatureResult> = () => {
   onInit?: HooksFactory<Input>;
   onDestroy?: HooksFactory<Input>;
-}): SignalStoreFeature<Input, EmptyFeatureResult> {
+};
+
+export function withHooks<Input extends SignalStoreFeatureResult>(
+  hooks:
+    | {
+        onInit?: HooksFactory<Input>;
+        onDestroy?: HooksFactory<Input>;
+      }
+    | HooksSupplier<Input>
+): SignalStoreFeature<Input, EmptyFeatureResult> {
```

Comment:
Instead of:

```ts
withHooks(() => ({
  onInit(store) {},
  onDestroy(store) {},
}))
```

Let's use:

```ts
withHooks((store) => ({
  onInit() {},
  onDestroy() {},
}))
```

I also suggest defining two signatures for the `withHooks` function instead of a union type for better readability, but also for easier potential deprecation of the first signature in the future.

Reply:
Done, since the old might become a candidate for deprecation, I named the new version's type `HookFactory` and the old one is now just a `HookFn`. Let me know, if should revert the renaming.